packages/cli/src/commands/hydrogen/dev.ts

```diff
@@ -156,3 +170,85 @@ async function runDev({
     },
   });
 }
+
+interface VariablesArguments {
+  root: string;
+  shop?: string;
+}
+async function combinedEnvironmentVariables({root, shop}: VariablesArguments) {
+  const remoteEnvironmentVariables = await pullRemoteEnvironmentVariables({
+    root,
+    flagShop: shop,
+    silent: true,
+  });
+
+  const formattedRemoteVariables = remoteEnvironmentVariables?.reduce(
+    (a, v) => ({...a, [v.key]: v.value}),
+    {},
+  );
+
+  const dotEnvPath = path.resolve(root, '.env');
```

Comment:
If there are no remote variables because the user hasn't linked a store, can we return `undefined` around here? That will allow existing users (who don't know about the link commands) to keep the existing behavior of refreshing worker when updating `.env`. And also won't change their logs until we get feedback about UX.

<img width="462" alt="image" src="https://github.com/Shopify/hydrogen/assets/1634092/737b908c-5502-4be6-abc9-977496542e50">

I'll try to implement the .env reload behavior in a different way in the meantime 👍

Reply:
[Fixed](https://github.com/Shopify/hydrogen/compare/f117de53c3d65bac0d7d5436bef91128607868ce..a611598f53038efb07c2405e200b8b50eac40f90#diff-59500481d90faf3dd4fa6e0c61a5a927ea1d65267e6788074793fe81af71e078R89-R92) by the same thing that was causing the rogue auth requets.